### DISCUSSION TITLE
How do I tell what color I'm playing?

### DISCUSSION THREAD

**Discussion by Austin-Nicholas** [SebLague/Chess-Challenge#388] [General]
Obviously there's things like IsWhiteToMove, and I can look at the board. But how does my bot know what it's supposed to optimize for?

**Comment by jongdetim** [comment]
By calling IsWhiteToMove, as you already knew. What is your question otherwise?

**Reply by jongdetim** [reply]
I call:
int color = board.IsWhiteToMove ? 1 : -1;
at the start of Think()

**Reply by Austin-Nicholas** [reply]
Thanks! I'm pretty new at programming, so I didn't put together that I could only call IsWhiteToMove on my own turn. 
I don't know why I didn't cause I ended up checking if plycount was even, so I guess I could have extrapolated that.
